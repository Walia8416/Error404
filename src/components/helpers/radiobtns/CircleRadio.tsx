import {Image, TouchableOpacity} from 'react-native';

import Images from '../../../constants/icon';
import React from 'react';

interface props {
  disable?: boolean;
  check?: boolean;
  setCheck?: () => void;
}

const CircleRadio = ({disable, check, setCheck}: props) => {
  return (
    <TouchableOpacity
      disabled={disable}
      onPress={setCheck}
      testID={'cicleradio'}
      style={{
        width: 16,
        height: 16,
        borderRadius: 8,
        justifyContent: 'center',
        alignItems: 'center',
        borderWidth: 1,
        borderColor: '#c4d7e7',
      }}>
      {check ? (
        <Image
          style={{width: 16, height: 16, borderRadius: 8}}
          source={Images.whiteRadio}
        />
      ) : (
        <Image
          style={{width: 16, height: 16, borderRadius: 8}}
          source={Images.greenRadio}
        />
      )}
    </TouchableOpacity>
  );
};

export default CircleRadio;
