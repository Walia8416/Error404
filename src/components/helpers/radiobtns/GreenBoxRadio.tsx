import {Image, TouchableOpacity, View} from 'react-native';

import React from 'react';
import Images from '../../../constants/icon';

interface props {
  check: boolean;
  onPress?: () => void;
  onPressIn?: () => void;
  onPressOut?: () => void;
}

const GreenBoxRadio = ({check, onPress, onPressIn, onPressOut}: props) => {
  // const [check, setCheck] = React.useState(false);

  return (
    <TouchableOpacity
      onPress={onPress ? onPress : () => undefined}
      onPressIn={onPressIn ? onPressIn : () => undefined}
      onPressOut={onPressOut ? onPressOut : () => undefined}
      testID="greenboxradio"
      style={{
        width: 16,
        height: 16,
        justifyContent: 'center',
        alignItems: 'center',
        borderWidth: 1,
        borderColor: '#c4d7e7',
      }}>
      {check ? (
        <Image style={{width: 16, height: 16}} source={Images.greenCheckbox} />
      ) : (
        <View style={{width: 16, height: 16, backgroundColor: '#ffffff'}} />
      )}
    </TouchableOpacity>
  );
};

export default GreenBoxRadio;
