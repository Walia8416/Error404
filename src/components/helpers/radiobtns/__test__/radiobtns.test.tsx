import {act, create} from 'react-test-renderer';

import CircleRadio from '../CircleRadio';
import GreenBoxRadio from '../GreenBoxRadio';
import React from 'react';

describe('radio btns test', () => {
  it(' circle radio btn test', () => {
    const circleradiobtn = create(<CircleRadio />);
    expect(circleradiobtn).toMatchSnapshot();
  });
  it(' box radio btn test', () => {
    const boxradiobtn = create(<GreenBoxRadio check={true} />);
    const boxbtn = boxradiobtn.root.findByProps({testID: 'greenboxradio'})
      .props;
    act(() => boxbtn.onPress());
    act(() => boxbtn.onPressIn());
    act(() => boxbtn.onPressOut());
  });
});
