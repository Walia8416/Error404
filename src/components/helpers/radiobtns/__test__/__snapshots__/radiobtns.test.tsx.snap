// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio btns test  circle radio btn test 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#c4d7e7",
      "borderRadius": 8,
      "borderWidth": 1,
      "height": 16,
      "justifyContent": "center",
      "opacity": 1,
      "width": 16,
    }
  }
  testID="cicleradio"
>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": 16,
        "width": 16,
      }
    }
  />
</View>
`;
